unit/ansi-c/max_malloc_size.cpp

```diff
@@ -0,0 +1,50 @@
+/*******************************************************************\
+
+Module: Unit test for max_malloc_size
+
+Author: Thomas Kiley
+
+\*******************************************************************/
+
+#include <ansi-c/ansi_c_internal_additions.cpp>
+#include <testing-utils/use_catch.h>
+
+TEST_CASE(
+  "max_malloc_size",
+  "[core][ansi-c][ansi_c_internal_additions][max_malloc_size]")
+{
+  cbmc_invariants_should_throwt throw_invariants;
+
+  SECTION("Too many bits for pointer ID")
+  {
+    REQUIRE_THROWS_AS(max_malloc_size(4, 4), invariant_failedt);
+    REQUIRE_THROWS_AS(max_malloc_size(4, 5), invariant_failedt);
+  }
+
+  SECTION("Not enough bits for pointer ID")
+  {
+    REQUIRE_THROWS_AS(max_malloc_size(4, 0), invariant_failedt);
+  }
+
+  SECTION("Not enough bits in the pointer")
+  {
+    REQUIRE_THROWS_AS(max_malloc_size(0, 0), invariant_failedt);
+  }
+
+  SECTION("Max allocation size overflow")
+  {
+  }
```

Comment:
Gah - fixed this locally and evidently didn't push :facepalm: will raise a subsequent PR...

Reply:
#5317